python/tests/sleeper/test_query.py

```diff
@@ -0,0 +1,164 @@
+    # Given
+    table_name = "test-table"
+    query_id = "test-query"
+    keys = {"key": ["my_key", "my_key2"]}
+    mocked_results = ["mocked row 1", "mocked row 2"]
+    mock_process = mocker.patch("sleeper.web_socket_query.WebSocketQueryProcessor.process_query", autospec=True, return_value=mocked_results)
+
+    # When
+    # We don't care about the results for this test as they are mocked.
+    await sleeper_client.web_socket_exact_key_query(table_name=table_name, keys=keys, query_id=query_id)
+
+    expected_queries = [
+        WebSocketQuery(table_name=table_name, query_id=query_id, key="key", max_value="my_key", min_value="my_key", strings_base64_encoded=False).to_json(),
+        WebSocketQuery(table_name=table_name, query_id=query_id, key="key", max_value="my_key2", min_value="my_key2", strings_base64_encoded=False).to_json(),
+    ]
+
+    actual_queries = [(call.args[-1]).to_json() for call in mock_process.call_args_list]
+
+    # Then
+    assert actual_queries == expected_queries
+
+
+@pytest.mark.asyncio
+async def should_send_range_query_with_client_via_web_socket(sleeper_client: SleeperClient, mocker: MockerFixture):
+    # Given
+    table_name = "test-table"
+    query_id = "test-query"
+    keys = [{"key": {"min": "a", "max": "z"}}]
+    mocked_results = ["mocked row 1", "mocked row 2"]
+    mock_process = mocker.patch("sleeper.web_socket_query.WebSocketQueryProcessor.process_query", autospec=True, return_value=mocked_results)
+
+    # When
+    # We don't care about the results for this test as they are mocked.
+    await sleeper_client.web_socket_range_key_query(table_name=table_name, keys=keys, query_id=query_id)
+
+    expected_queries = [
+        WebSocketQuery(table_name=table_name, query_id=query_id, key="key", max_value="z", min_value="a", strings_base64_encoded=False).to_json(),
+    ]
+
+    actual_queries = [(call.args[-1]).to_json() for call in mock_process.call_args_list]
+
+    # Then
+    assert actual_queries == expected_queries
+
+
+@pytest.fixture
+def sleeper_client(properties: InstanceProperties) -> SleeperClient:
+    LocalStack.create_bucket(properties.get(CommonCdkProperty.CONFIG_BUCKET))
+    save_instance_properties(LocalStack.s3_resource(), properties)
+    return LocalStackSleeperClient.create(properties)
+
+
+@pytest.fixture
+def properties(query: Queue, table: Table, bucket: Bucket) -> InstanceProperties:
+    properties = create_test_instance_properties()
+    properties.set(QueryCdkProperty.QUERY_QUEUE_URL, query.url)
+    properties.set(QueryCdkProperty.QUERY_TRACKER_TABLE, table.table_name)
+    properties.set(QueryCdkProperty.QUERY_RESULTS_BUCKET, bucket.name)
+    properties.set(QueryCdkProperty.QUERY_WEBSOCKET_URL, "ws://localhost:7777/_aws/ws/api_id/test")
+    properties.set(CommonCdkProperty.REGION, "eu-west-2")
+    return properties
+
+
+@pytest.fixture
+def query() -> Queue:
+    return LocalStack.create_queue()
```

Comment:
The name for this fixture makes the test look like you're passing a query object into the "receive_messages" method, when it's actually the queue resource.

Reply:
It's still called query. It's the query queue.